umongo/data_proxy.py

```diff
@@ -145,7 +145,8 @@ def get(self, name, to_raise=KeyError):
     def set(self, name, value, to_raise=KeyError):
         name, field = self._get_field(name, to_raise)
         value = field._deserialize(value, name, None)
-        field._validate(value)
+        if getattr(field, 'allow_none', False) is False or value is not None:
```

Comment:
small optim: I would switch the checks given None compare is really cheaper than `getattr`:
` if value is not None or getattr(field, 'allow_none', False) is False:`